gtsam/geometry/Point3.cpp

```diff
@@ -75,6 +75,19 @@ double dot(const Point3 &p, const Point3 &q, OptionalJacobian<1, 3> H1,
   return p.x() * q.x() + p.y() * q.y() + p.z() * q.z();
 }
 
+Point3Pair mean(const std::vector<Point3Pair> &abPointPairs) {
+  const size_t n = abPointPairs.size();
+  Point3 aCentroid(0, 0, 0), bCentroid(0, 0, 0);
+  for (const Point3Pair &abPair : abPointPairs) {
+    aCentroid += abPair.first;
+    bCentroid += abPair.second;
+  }
+  const double f = 1.0 / n;
```

Comment:
Throw an exception if `n==0`?
PS: Sorry for the late comment... :-S

Reply:
@jlblancoc Thanks for the catch. I will create a new PR to fix it.